beamer/tests/agent/test_request.py

```diff
@@ -147,12 +153,14 @@ def test_fill_and_claim(request_manager, token, agent, allow_unlisted_pairs):
     assert claim.claimer == agent.address
 
 
-def test_withdraw(request_manager, token, agent):
+def test_withdraw(request_manager, token, agent, direction):
     requester, target = alloc_accounts(2)
     request_id = make_request(request_manager, token, requester, target, 1)
 
     with Sleeper(10) as sleeper:
-        while (request := agent.context.requests.get(request_id)) is None:
+        while (
+            request := agent._event_processors[direction]._context.requests.get(request_id)
+        ) is None:
             sleeper.sleep(0.1)
```

Comment:
This is not nice, in addition to accessing private variables.
How about something like

```python
   context = agent.get_context(direction)
   while (request := context.requests.get(request_id)) is None:
      sleeper.sleep(0.1)
```
?